src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java

```diff
@@ -197,7 +197,10 @@ else if ("none".equals(referencesPrefixPref))
             {
                String filename = FileSystemItem.createFile(docPath).getStem();
                PanmirrorUIToolsAttr attr = new PanmirrorUITools().attr;
-               options.references.prefix = attr.pandocAutoIdentifier("ref_" + filename) + "-";
+               String identifier = attr.pandocAutoIdentifier(filename);
+               if (identifier.length() == 0)
+                  identifier = attr.pandocAutoIdentifier("ref-" + filename);
```

Comment:
Should this be `ref_` instead of `ref-`? (to match the prior behavior)

Reply:
Ultimately went for artificially adding "a" so that `pandocAutoIdentifier()` does not trim at the left, but then remove it, so that there's no artificial prefix.

footnotes from `01.Rmd` become `01-1` etc ... and footnote from e.g. `file3.Rmd` become `file3-1` etc ...